pkg/kudoctl/cmd/init.go

```diff
@@ -0,0 +1,155 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	manInit "github.com/kudobuilder/kudo/pkg/kudoctl/cmd/init"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kube"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+const initDesc = `
+This command installs KUDO Manager onto your Kubernetes Cluster and sets up local configuration in $KUDO_HOME (default ~/.kudo/).
+
+As with the rest of the KUDO commands, 'kudo init' discovers Kubernetes clusters
+by reading $KUBECONFIG (default '~/.kube/config') and using the default context.
+
+To set up just a local environment, use '--client-only'. That will configure
+$KUDO_HOME, but not attempt to connect to a Kubernetes cluster and install the KUDO Manager.
+
+When installing KUDO manager, 'kudo init' will attempt to install the latest released
+version. You can specify an alternative image with '--kudo-image'.
+
+To dump a manifest containing the KUDO deployment YAML, combine the
+'--dry-run' and '--debug' flags.
+`
+
+type initCmd struct {
+	out        io.Writer
+	fs         afero.Fs
+	image      string
+	dryRun     bool
+	output     string
+	version    string
+	wait       bool
+	timeout    int64
+	kubeClient kubernetes.Interface
+}
+
+func newInitCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	i := &initCmd{fs: fs, out: out}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initialize KUDO on both client and server",
+		Long:  initDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("this command does not accept arguments")
+			}
+
+			return i.run()
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&i.image, "kudo-image", "i", "", "Override KUDO manager image and/or version")
+	f.StringVarP(&i.version, "version", "", "", "Override KUDO manager version of the kudo image")
+	f.StringVarP(&i.output, "output", "o", "", "Output format")
+	f.BoolVar(&i.dryRun, "dry-run", false, "Do not install local or remote")
+	f.BoolVarP(&i.wait, "wait", "w", false, "Block until KUDO manager is running and ready to receive requests")
+	f.Int64Var(&i.timeout, "wait-timeout", 300, "Wait timeout to be used")
+
+	return cmd
+}
+
+// run initializes local config and installs KUDO manager to Kubernetes cluster.
+func (i *initCmd) run() error {
+	opts := manInit.NewOptions(i.version)
+	// if image provide switch to it.
+	if i.image != "" {
+		opts.Image = i.image
+	}
+
+	//TODO: implement output=yaml|json (define a type for output to constrain)
+	//define an Encoder to replace YAMLWriter
+	if strings.ToLower(i.output) == "yaml" {
```

Comment:
I think in k8s world, using -o actually means that nothing is executed on the server, so we should exit after we print it out

Reply:
okay, I was wrong